source/extensions/quic_listeners/quiche/platform/quic_logging_impl.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <cerrno>
+#include <cstring>
+#include <iostream>
+#include <sstream>
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+#include "absl/base/optimization.h"
+
+#define QUIC_LOG_IMPL_INTERNAL(condition, logstream)                                               \
```

Comment:
A brief comment explaining roughly how this works, and that it is done in order to instantiate logstream only if condition is true, would be nice.

Side question: I was noticing that base/logging.h does this slightly differently, placing the log stream expression in the default case of a switch(0), do you have any idea why that is? I agree your formulation seems simpler, just wondering if there's some reason behind the base/logging.h form.

Reply:
Comment added.

I was following the style in go/onelog, not base/logging.h. But the style in go/onelog no longer compiles after I updated my clang to 7.0, so I've switched to the base/logging.h style.

About the switch(0):
If a macro is implemented in the form of "if (x) {} else {}", the switch(0) can be used to suppress a warning about ambiguous else blocks. Suppose there is

#define MACRO()  if (x) {} else {}

And a call site:

if (y) MACRO();

Which expands to:

if (y) if (x) {} else {};

The "else" is bind to the second "if", but some compiler worries that the intent is to bind to the first "if", so they give warnings. The switch(0) trick eliminates this warning.